qa/Rakefile

```diff
@@ -24,10 +30,10 @@ namespace :qa do
     desc "Bootstrap all the VM's used for this tests"
     task :setup, :platform do |t, args|
       config   = PlatformConfig.new
-      machines = config.select_names_for(args[:platform])
+      experimental = (ENV['LS_QA_EXPERIMENTAL_OS'].to_s.downcase || "false") == "true"
+      machines = config.select_names_for(args[:platform], {"experimental" => experimental})
```

Comment:
My thought on using an env var was because it changes behaviour of same
task, run default or experimental ones. It makes the interface a bit more
clear to me. But i'm good with whatever we agree with.

On Thu, 16 Jun 2016 19:10 Pier-Hugues Pellerin, <notifications@github.com>
wrote:

> In qa/Rakefile
> <https://github.com/elastic/logstash/pull/5506#discussion_r67385378>:
>
> > @@ -24,10 +30,10 @@ namespace :qa do
> >      desc "Bootstrap all the VM's used for this tests"
> >      task :setup, :platform do |t, args|
> >        config   = PlatformConfig.new
> > -      machines = config.select_names_for(args[:platform])
> > +      experimental = (ENV['LS_QA_EXPERIMENTAL_OS'].to_s.downcase || "false") == "true"
> > +      machines = config.select_names_for(args[:platform], {"experimental" => experimental})
>
> We are using rake task arguments and an environment variable in the same
> task can we consolidate in only one way?
> Since we are using rake arguments I think we should use that for the
> experimental option? Opinions?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/elastic/logstash/pull/5506/files/fa7650fd919a983f62bb51e0ce32f5f4cc7078e3#r67385378>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAELvBLSODW_4dG0IIB14VNde4vUwmgYks5qMYOegaJpZM4I3Njq>
> .
>